DocumentService/Repositories/DocumentRepository.cs

```diff
@@ -18,20 +19,39 @@ public DocumentRepository(DocumentContext context)
         {
             this.context = context;
         }
+        
+        public async Task<bool> GetCorrelationById(Guid id)
+        {
+            var query = from q in context.Correlation
+                        where q.CorrelationId == id
+                        select q;
+
+            return await query.AnyAsync();
+        }
+        public async Task<DocumentInfo> GetDocumentByCorrelationId(Guid id)
+        {
+            var query = from q in context.DocumentInfo
+                        where q.CorrelationId == id
+                        select q;
+
+            return await query.FirstOrDefaultAsync();
+        }
```

Comment:
I don't believe we need this method, but if we want to keep it change it to:
```suggestion
        public async Task<DocumentInfo> GetDocumentByCorrelationId(Guid correlationId)
        {
            var query = this.Filter(x => x.CorrelationId == correlationId);
            return await query.FirstOrDefaultAsync();
        }
```